buildtools/stack/stack_test.go

```diff
@@ -0,0 +1,101 @@
+package stack
```

Comment:
Ideally we should work towards creating external tests under a package like `stack_test`. You'll see this pattern elsewhere in the codebase. The reason we want to do that is to ensure the focus of the test is on the external contract of the public methods. We shouldn't really care how the internals behave.

Reply:
Whoops -- didn't know that convention. I'll fix this